ci-operator/step-registry/ibmcloud/provision/vpc/mult-subnets/ibmcloud-provision-vpc-mult-subnets-commands.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+MORE_SUBNETS_COUNT=$((EXTRA_SUBNETS_NUMBER))
+#15 is the max number of of ALB, so just create more subnets on vpc, do not use them when create cluster. https://cloud.ibm.com/docs/vpc?topic=vpc-load-balancer-faqs#max-number-subnets-alb
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output   
+    region="${LEASED_RESOURCE}"
+    export region
+    "${IBMCLOUD_CLI}" config --check-version=false
+    echo "Try to login..." 
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+function waitAvailable() {
+    local retries=15  try=0 
+    local type="$1" name="$2"
+    while [ "$(${IBMCLOUD_CLI} is ${type} ${name} --output JSON | jq -r '.status')" != "available" ] && [ $try -lt $retries ]; do
+        echo "The ${name} is not available, waiting..."
+        sleep 10
+        try=$(expr $try + 1)
+    done
+
+    if [ X"$try" == X"$retries" ]; then
+        echo "Fail to get available ${type} - ${name}"
+        "${IBMCLOUD_CLI}" is ${type} ${name} --output JSON
+        return 1
+    fi   
+}
+
+function create_subnets() {
```

Comment:
Why not reuse `ibmcloud-provision-vpc` to create the wanted number of subnets ?

Reply:
add env parameters in ibmcloud-provision-vpc? need not the separate step ibmcloud-provision-vpc-mult-subnets?

Reply:
> add env parameters in ibmcloud-provision-vpc? need not the separate step ibmcloud-provision-vpc-mult-subnets

Yeah. 


I think the logic to update the existing ibmcloud-provision-vpc step is clear:
1. Introduce two ENV var, e.g: NUM_SUBNETS (default: empty), NUM_SUBNETS_USED  (default: empty)
2. when NUM_SUBNETS==empty, do the same thing as before, when NUM_SUBNETS_USED==empty, do the same thing as before.
3. when NUM_SUBNETS != empty, create the subnet in the order of zone one by one, until reached the wanted number.
4. when NUM_SUBNETS_USED != empty, save the wanted number subnet info.

Reply:
how about the default value as the following?
NUM_SUBNETS=6
NUM_SUBNETS_USED=6

Reply:
1. How about changing the NUM_SUBNETS_USED to USE_ALL (boolean)?  
When empty or "no", use 6 subnets as previously.  
When "yes", use all the created subnets in the cluster. 
That will be easier when creating the customer_vpc_subnets.yaml

2. I prefer using a separate method to create the extra subnets. 
When verifying [OCPBUGS-36236](https://issues.redhat.com/browse/OCPBUGS-36236), need to use the subnet on the second page  (the default list is ordered by the create time), and the subnets used in the cluster need to cover every zone.  
When creating all the subnets together, hard to select the subnets to use in the cluster.

Reply:
> How about changing the NUM_SUBNETS_USED to USE_ALL (boolean)?

How to cover creating 51 subnets, only use 3 of them in install-config ?

> I prefer using a separate method to create the extra subnets.

Selecting the target subnets is easy, using reverse order, or random way, you can control the behavior by introducing a new ENV var.

Reply:
for cover the [OCP-77280](https://polarion.engineering.redhat.com/polarion/redirect/project/OSE/workitem?id=OCP-77280) [OCP-77281](https://polarion.engineering.redhat.com/polarion/redirect/project/OSE/workitem?id=OCP-77281) [OCP-77282](https://polarion.engineering.redhat.com/polarion/redirect/project/OSE/workitem?id=OCP-77282)
Using the USED_IN_CLUSTER (USE_ALL) boolean is enough 
creating 51 subnets, use 6 subnets as default.  

Merge creating the extra subnets into one step, preparing the customer_vpc_subnets.yaml will be difficult, I failed to find a solution.

Reply:
How to get the subnet name list from [vpc_info](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/58092/rehearse-58092-periodic-ci-openshift-verification-tests-master-installation-nightly-4.16-ibmcloud-ipi-mult-subnets-f28/1852245579111862272/artifacts/ibmcloud-ipi-mult-subnets-f28/ibmcloud-provision-vpc-mult-subnets/artifacts/vpc_info) 
order by "created_at" filed.  then get 3 named 'control-plane' 3 named 'compute'
or based on the output of  _ibmcloud is subnets --output JSON | jq 'sort_by(.created_at)'_

Reply:
I opened https://github.com/openshift/release/pull/58397 to support a large number of IBMCLoud vpc subnets, once it get merged, you can reference it in your install workflow.